File: ue1/a2.ts
const k = 100;

function sieveOfEratosthenes(k: number): number[] {
  let prime: boolean[] = new Array(k + 1).fill(true);
  let primeNumbers: number[] = [];

  for (let p = 2; p <= k; p++) {
    if (prime[p]) {
      primeNumbers.push(p);
      // Mark all multiples of p as false
      for (let i = p * p; i <= k; i += p) {
        prime[i] = false;
      }
    }
  }
  return primeNumbers;
}

console.log("Prime numbers from 1 to", k, ":", sieveOfEratosthenes(k));
console.log(
  "Total prime numbers from 1 to",
  k,
  ":",
  sieveOfEratosthenes(k).length
);

// performance in milliseconds
console.log("From 1 to 100000", sieveOfEratosthenes(100000), performance.now());

File: ue1/a2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> sieveOfEratosthenes(int n)
{
    vector<bool> prime(n + 1, true);
    vector<int> primeNumbers;

    for (int p = 2; p <= n; p++)
    {
        if (prime[p])
        {
            primeNumbers.push_back(p);
            for (int i = p * p; i <= n; i += p)
            {
                prime[i] = false;
            }
        }
    }
    return primeNumbers;
}

int main()
{
    int n = 100;
    vector<int> primeNumbers = sieveOfEratosthenes(n);
    for (int i = 0; i < primeNumbers.size(); i++)
    {
        cout << primeNumbers[i] << " ";
    }
    return 0;
}